lib/src/algorithms.dart

```diff
@@ -32,10 +32,10 @@ int _comparableBinarySearch/*<T extends Comparable<T>>*/(
 /// the objects.
 ///
 /// Returns -1 if [value] is not in the list by default.
-int binarySearch/*<T extends Comparable<T>>*/(
+int binarySearch/*<T>*/(
     List/*<T>*/ sortedList, /*=T*/ value, { int compare(/*=T*/ a, /*=T*/ b) }) {
   if (compare == null) {
-    return _comparableBinarySearch(sortedList, value);
+    return _comparableBinarySearch(sortedList as List<Comparable>, value as Comparable);
```

Comment:
Just a drive by comment:  you guys (flutter) are mostly using strong mode just for the static checking, but packages like this will also be used with DDC, which injects a runtime checking component.  So we want to be sure that any `as` checks that are inserted are compatible with the DDC runtime checks in the expected use cases.  I don't know what contract is expected of this function: but, if it is part of the contract that `binarySearch<dynamic>(<dynamic>[xxxx], x)` should work, where `xxx` is a series of things that implement `Comparable`.... then that cast is wrong, since on DDC it will fail.  If the contract is that only lists which were created as `List<Comparable>` (or subtypes) are allowed, then the cast is ok.